src/web/api/src/main/java/org/locationtech/geogig/rest/geotools/SimpleFeatureStoreRepresentation.java

```diff
@@ -0,0 +1,83 @@
+/* Copyright (c) 2016 Boundless and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/org/documents/edl-v10.html
+ *
+ * Contributors:
+ * Erik Merkle (Boundless) - initial implementation
+ */
+package org.locationtech.geogig.rest.geotools;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.concurrent.ExecutionException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.geotools.data.simple.SimpleFeatureStore;
+import org.locationtech.geogig.api.AbstractGeoGigOp;
+import org.locationtech.geogig.geotools.plumbing.ExportOp;
+import org.locationtech.geogig.rest.AsyncCommandBinaryRepresentation;
+import org.locationtech.geogig.rest.AsyncContext.AsyncCommand;
+import org.locationtech.geogig.rest.CommandBinaryRepresentationFactory;
+import org.restlet.data.MediaType;
+
+import com.google.common.io.ByteStreams;
+
+/**
+ * Representation for asynchronous commands with SimpleFeatureStore results AND a binary file
+ * that contains the result that can be streamed back to the consumer.
+ */
+public class SimpleFeatureStoreRepresentation extends AsyncCommandBinaryRepresentation<SimpleFeatureStore> {
+
+    protected final File binary;
+
+    public SimpleFeatureStoreRepresentation(MediaType mediaType,
+        AsyncCommand<SimpleFeatureStore> cmd, File binary) {
+        super(mediaType, cmd);
+        this.binary = binary;
+    }
+
+    @Override
+    public void write(OutputStream outputStream) {
+        try (FileInputStream fis = new FileInputStream(binary)) {
+            // Block until get() returns. We don't actually do anything with the result, we just
+            // need to wait for the operation to complete. Once complete, we can stream the binary
+            // file back.
+            // NOTE: This will not work if the command takes longer than the TCP timeout to
```

Comment:
@groldan This is the part I am most concerned with. This code will block waiting for the GeoPkg export to complete (on the server) before it starts to write the export file out to the client stream. If the export command takes too long to complete, the stream will close due to no data. Ultimately, this should probably return a *retrieval URL* in an XML/JSON response that consuming clients can poll and retrieve on their own.